extensions/amazon-lambda-rest/runtime/src/main/java/io/quarkus/amazon/lambda/http/CustomPrincipal.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.amazon.lambda.http;
+
+import java.security.Principal;
+import java.util.Map;
+
+/**
+ * Allocated when a custom authorizer (i.e. Lambda) is used to authenticate user
+ *
+ * Will only be allocated if requestContext.authorizer.principalId is set
+ * in the http event sent by API Gateway
+ *
+ */
+public class CustomPrincipal implements Principal {
```

Comment:
This is a bit subjective, but did you consider just using the attributes on QuarkusSecurityIdentity instead of a custom Principal?

Reply:
The map obtained from Lambda custom security is arbitrary.  Was just going to let the user deal with this and write their own custom IDP if they want to propagate anything.

Reply:
We already have a custom map though at io.quarkus.security.identity.SecurityIdentity#getAttributes